### PR TITLE
(#2258) Linux pack all files if no files element in nuspec

### LINKED ISSUE #2258: Packing on Mono on Linux will not include any files in .nupkg if there is no files element in the .nuspec
### What You Are Seeing?

On Mono on Linux.

If there is no `<files>` element in the `.nuspec`, then when running `choco pack`, no files apart from the `.nuspec` will be included in the `.nupkg`. 

### What is Expected?

On Windows, when running `choco pack` when the `.nuspec` has no `<files>` element, then all files and folder from the directory that the `.nuspec` is in will be included.

I expect that the behavior between the two platforms is the same.  I don't care which behavior it is, just that it is the same,

### How Did You Get This To Happen? (Steps to Reproduce)

1. Create or copy the source for a package to a Linux system with choco
2. Completely remove the  `<files>` element from the `.nuspec`
3. Run `choco pack`
4. Inspect the resulting `.nupkg` and notice that there are no files or folders included. (E.G. the tools directory is missing)

### Output Log

<details>
<summary>Full Log Output</summary>

<p>

~~~sh
2021-03-05 21:49:24,075 31299 [DEBUG] - XmlConfiguration is now operational
2021-03-05 21:49:24,135 31299 [INFO ] - ============================================================
2021-03-05 21:49:24,635 31299 [INFO ] - Chocolatey v0.10.16-beta-116-gb6f49599
2021-03-05 21:49:24,639 31299 [DEBUG] - Chocolatey is running on Linux v 5.10.0.0
2021-03-05 21:49:24,640 31299 [DEBUG] - Attempting to delete file "/opt/chocolatey/chocolatey.dll.old".
2021-03-05 21:49:24,640 31299 [DEBUG] - Attempting to delete file "/opt/chocolatey/choco.exe.old".
2021-03-05 21:49:24,650 31299 [DEBUG] - Command line: /opt/chocolatey/choco.exe pack --verbose --debug --allow-unofficial
2021-03-05 21:49:24,651 31299 [DEBUG] - Received arguments: pack --verbose --debug --allow-unofficial
2021-03-05 21:49:24,690 31299 [DEBUG] - RemovePendingPackagesTask is now ready and waiting for PreRunMessage.
2021-03-05 21:49:24,700 31299 [DEBUG] - Sending message 'PreRunMessage' out if there are subscribers...
2021-03-05 21:49:24,703 31299 [DEBUG] - [Pending] Removing all pending packages that should not be considered installed...
2021-03-05 21:49:24,742 31299 [DEBUG] - Performing validation checks.
2021-03-05 21:49:24,745 31299 [DEBUG] - Global Configuration Validation Checks:
2021-03-05 21:49:24,746 31299 [DEBUG] -  - Package Exit Code / Exit On Reboot = Checked
2021-03-05 21:49:24,746 31299 [DEBUG] - System State Validation Checks:
2021-03-05 21:49:24,749 31299 [INFO ] - 2 validations performed. 2 success(es), 0 warning(s), and 0 error(s).
2021-03-05 21:49:24,749 31299 [DEBUG] - The source 'https://chocolatey.org/api/v2/' evaluated to a 'normal' source type
2021-03-05 21:49:24,750 31299 [DEBUG] -
NOTE: Hiding sensitive configuration data! Please double and triple
 check to be sure no sensitive data is shown, especially if copying
 output to a gist for review.
2021-03-05 21:49:24,767 31299 [DEBUG] - Configuration: CommandName='pack'|CacheLocation='/tmp/chocolatey'|
ContainsLegacyPackageInstalls='True'|
CommandExecutionTimeoutSeconds='2700'|WebRequestTimeoutSeconds='30'|
Sources='https://chocolatey.org/api/v2/'|SourceType='normal'|
Debug='True'|Verbose='True'|Trace='False'|Force='False'|Noop='False'|
HelpRequested='False'|UnsuccessfulParsing='False'|RegularOutput='True'|
QuietOutput='False'|PromptForConfirmation='True'|AcceptLicense='False'|
AllowUnofficialBuild='True'|AllVersions='False'|
SkipPackageInstallProvider='False'|Prerelease='False'|ForceX86='False'|
OverrideArguments='False'|NotSilent='False'|
ApplyPackageParametersToDependencies='False'|
ApplyInstallArgumentsToDependencies='False'|IgnoreDependencies='False'|
AllowMultipleVersions='False'|AllowDowngrade='False'|
ForceDependencies='False'|Information.PlatformType='Linux'|
Information.PlatformVersion='5.10.0.0'|
Information.PlatformName='Linux'|
Information.ChocolateyVersion='0.10.16.0'|
Information.ChocolateyProductVersion='0.10.16-beta-116-gb6f49599'|
Information.FullName='chocolatey, Version=0.10.16.0, Culture=neutral, PublicKeyToken=fd112f53c3ab578c'|

Information.Is64BitOperatingSystem='True'|
Information.Is64BitProcess='True'|Information.IsInteractive='False'|
Information.UserName=redacted
Information.IsUserAdministrator='False'|
Information.IsUserSystemAccount='False'|
Information.IsUserRemoteDesktop='False'|
Information.IsUserRemote='True'|
Information.IsProcessElevated='False'|
Information.IsLicensedVersion='False'|Information.LicenseType='Foss'|
Information.CurrentDirectory='pkg-testing/no-files-zfsin'|

Features.AutoUninstaller='True'|Features.ChecksumFiles='True'|
Features.AllowEmptyChecksums='False'|
Features.AllowEmptyChecksumsSecure='True'|
Features.FailOnAutoUninstaller='False'|
Features.FailOnStandardError='False'|Features.UsePowerShellHost='True'|
Features.LogEnvironmentValues='False'|Features.LogWithoutColor='False'|
Features.VirusCheck='False'|
Features.FailOnInvalidOrMissingLicense='False'|
Features.IgnoreInvalidOptionsSwitches='True'|
Features.UsePackageExitCodes='True'|
Features.UseEnhancedExitCodes='False'|
Features.UseFipsCompliantChecksums='False'|
Features.ShowNonElevatedWarnings='True'|
Features.ShowDownloadProgress='True'|
Features.StopOnFirstPackageFailure='False'|
Features.UseRememberedArgumentsForUpgrades='False'|
Features.IgnoreUnfoundPackagesOnUpgradeOutdated='False'|
Features.SkipPackageUpgradesWhenNotInstalled='False'|
Features.RemovePackageInformationOnUninstall='False'|
Features.ExitOnRebootDetected='False'|
Features.LogValidationResultsOnWarnings='True'|
Features.UsePackageRepositoryOptimizations='True'|
Features.ScriptsCheckLastExitCode='False'|
ListCommand.LocalOnly='False'|
ListCommand.IdOnly='False'|ListCommand.IncludeRegistryPrograms='False'|
ListCommand.PageSize='25'|ListCommand.Exact='False'|
ListCommand.ByIdOnly='False'|ListCommand.ByTagOnly='False'|
ListCommand.IdStartsWith='False'|ListCommand.OrderByPopularity='False'|
ListCommand.ApprovedOnly='False'|
ListCommand.DownloadCacheAvailable='False'|
ListCommand.NotBroken='False'|
ListCommand.IncludeVersionOverrides='False'|
UpgradeCommand.FailOnUnfound='False'|
UpgradeCommand.FailOnNotInstalled='False'|
UpgradeCommand.NotifyOnlyAvailableUpgrades='False'|
UpgradeCommand.ExcludePrerelease='False'|
NewCommand.AutomaticPackage='False'|
NewCommand.UseOriginalTemplate='False'|SourceCommand.Command='unknown'|
SourceCommand.Priority='0'|SourceCommand.BypassProxy='False'|
SourceCommand.AllowSelfService='False'|
SourceCommand.VisibleToAdminsOnly='False'|
FeatureCommand.Command='unknown'|ConfigCommand.Command='unknown'|
ApiKeyCommand.Remove='False'|PinCommand.Command='unknown'|
OutdatedCommand.IgnorePinned='False'|Proxy.BypassOnLocal='True'|
2021-03-05 21:49:24,767 31299 [DEBUG] - _ Chocolatey:ChocolateyPackCommand - Normal Run Mode _
2021-03-05 21:49:24,955 31299 [INFO ] - Attempting to build package from 'zfsin.nuspec'.
2021-03-05 21:49:25,090 31299 [INFO ] - Successfully created package 'pkg-testing/no-files-zfsin/zfsin.0.24-eljun.nupkg'
2021-03-05 21:49:25,091 31299 [DEBUG] - Sending message 'PostRunMessage' out if there are subscribers...
2021-03-05 21:49:25,093 31299 [DEBUG] - Exiting with 0
~~~

</p>

</details>

### PR DESCRIPTION
**Pull request by TheCakeIsNaOH**
Fixes https://github.com/chocolatey/choco/issues/2258

### CODE REVIEW

File: src/Core/Authoring/PackageBuilder.cs [hunk 1]
```diff
@@ -455,7 +455,7 @@ private void ReadManifest(Stream stream, string basePath, IPropertyProvider prop
             {
                 if (manifest.Files == null)
                 {
-                    AddFiles(basePath, @"**\*", null);
+                    AddFiles(basePath, string.Format("**{0}*",Path.DirectorySeparatorChar), null);
```
Comment (ferventcoder) [inline src/Core/Authoring/PackageBuilder.cs]: Would be good if this would go down a few levels of subdirectories as well

Comment (ferventcoder) [reply src/Core/Authoring/PackageBuilder.cs]: Or like all the way down subdirectories - I have not seen it get to a subdirectory under `tools` and pack that in. Which is unfortunate

Comment (ferventcoder) [reply src/Core/Authoring/PackageBuilder.cs]: The change looks good tho!

Comment (ferventcoder) [reply src/Core/Authoring/PackageBuilder.cs]: I mean the pattern looks lime it should work, but I've not seen it work properly in practice.

Repro: 
1. create a nuspec manifest without a files section. 
1. create a tools folder
2. In the tools folder, create a web folder
3. In the web folder, put a couple of files.
4. Put some files in the tools folder
5. Put some files in the top level folder.
6. run choco pack
7. Rename the nupkg to a zip
8. Unzip the nupkg and inspect what is in there.

Comment (TheCakeIsNaOH) [reply src/Core/Authoring/PackageBuilder.cs]: I'll take a look, and see if I can figure that out that issue as well. That is issue #20, correct?

On a tangent, I actually have 7zip set as the default file association for `.nupkg`s, so I can double-click the compiled package and explore the contents really quickly.

Comment (gep13) [reply src/Core/Authoring/PackageBuilder.cs]: @TheCakeIsNaOH where you able to further test the scenario that was described above?

Comment (TheCakeIsNaOH) [reply src/Core/Authoring/PackageBuilder.cs]: @gep13 I did test it, but was unable to replicate the issue..

Comment (gep13) [reply src/Core/Authoring/PackageBuilder.cs]: Okay, thanks for confirming.  I am going to move forward with pulling this in.

**Comment by gep13** [comment]
For future folks looking at this PR, the issue referenced in this PR, and in the commit message, references the issue on the chocolatey/choco repository.  For future PR's, we should replicate the issue in this repository, so that we can correctly hook everything up.

**Review by gep13 — APPROVED** [review]
LGTM!

**Comment by gep13** [comment]
@TheCakeIsNaOH thanks for getting this updated! Really appreciate it!

**Comment by TheCakeIsNaOH** [comment]
@gep13 Is the `.dll` in the choco repository going to updated with changes from this and #19 soon, or is updating it going to be held off for the more major nuget version upgrades that are scheduled?

**Comment by gep13** [comment]
It isn't in there yet. There is at least one more change that I need to make in here, and then my intention is to include a new version of the assembly in the next release of Chocolatey.

**Comment by TheCakeIsNaOH** [comment]
Ok.

My concern is more for when an updated assembly is planned to be put in the choco git repository, for people self building on mono.

**Comment by gep13** [comment]
>@TheCakeIsNaOH said...
My concern is more for when an updated assembly is planned to be put in the choco git repository, for people self building on mono.

Understood. If/when I get the PR that I just created in here merged in, I will be in a position to add the updated assembly to the chocolatey/choco repository, and you should be able to take it for a spin.

**Comment by gep13** [comment]
@TheCakeIsNaOH I have updated the NuGet.Core.dll that is used in chocolatey/choco stable branch to include the work that has been done in this repository.

**Pull request merged**